exporter/collector/googlemanagedprometheus/naming.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlemanagedprometheus
+
+import (
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func GetMetricName(baseName string, metric pdata.Metric) string {
+	return baseName + gmpMetricSuffix(metric)
+}
+
+func gmpMetricSuffix(metric pdata.Metric) string {
```

Comment:
What do we decide to do with `unknown` typed metrics?

Reply:
Otel currently does not have a notion of unknown typed metrics.  They are currently changed to gauges in the receiver.  Once otel adds support for them, we can handle them correctly here.  Supporting unknown-typed metrics is tracked here: https://github.com/open-telemetry/wg-prometheus/issues/69

Reply:
Note fixing this to handle unknown metrics properly is backward-incompatible and would require a major version bump. We should probably try to get it addressed before marking exporter support for Prometheus stable.

Reply:
Agreed.  The same goes for info + stateset types if those are added in the future (they become gauges right now).